engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -564,6 +583,22 @@ static void DestroyTextureImage(dmGraphics::TextureImage& texture_image)
     delete[] texture_image.m_Alternatives.m_Data;
 }
 
+static void CheckTextureResource(lua_State* L, int i, const char* field_name, const char** texture_path_out, dmGraphics::HTexture* texture_out)
+{
+    lua_getfield(L, i, field_name);
+    if (!lua_isstring(L, -1))
```

Comment:
I think in general, for our "CheckXxx" functions we use "CheckHashOrString()" for getting the name of something.

I think it's likely that we might want to be able to receive a hash here?

Reply:
E.g. if the user wants to use a texture that is assigned to some resource (e.g. a sprite).
It doesn't even have to be a dynamically generated texture.

Looking at the code below, we use the string to set it into the texture set.
But, perhaps we can leave that string as 0, and instead set a texture name hash?